### DISCUSSION TITLE
DataGrid sample not working for WinUI( Microsoft.UI.Xaml.Markup.XamlParseException)

### DISCUSSION THREAD

**Discussion by MasterThreadInf** [unoplatform/uno#11336] [Q&A]
Here is example I've created with the help of Unosample : https://github.com/MasterThreadInf/UNODataGrid.git
It is working for wasm ,android but not for Windows. The error I'm getting is  "Microsoft.UI.Xaml.Markup.XamlParseException". Can anyone help me to understand and solve this issue? 

Thanking in advance!

**Comment by MasterThreadInf** [comment]
I uderstood where it have problem , it is this.InitializeComponent(); in LoginStatus.Xaml.cs .  But why ? Incompatibility of frameworks or parsing things which can't be worked in WinUI ?

**Reply by jeromelaban** [reply]
You can take a look at the debugging output and you'll see messages like these:
```
WinRT information: Cannot find a Resource with the Name/Key WelcomeTextStyle [Line: 65 Position: 64]
```

**Reply by MasterThreadInf** [reply]
Well there comes many errors which aren't helpful at all like InitiliazeComponent  ,Binding, and many such things which actually are irrevelant errors. And this welcomeTextStyle seems like one of such error because if you look at resource you'll find Name/key
https://github.com/MasterThreadInf/UNODataGrid/blob/master/WCTTestbed2/WCTTestbed2.Shared/Resources/Styles.xaml. It becomes difficult to seperate real errror from this kind of error.  By the way ,code works absolutely fine for wasm when I unload WinUI or when I comment out above this.Initializecomponent ( by doing this WinUI also work).

**Reply by MasterThreadInf** [reply]
Its working now but when I try same thing using 
uno.extension_ template. It isn't working.

![errors](https://user-images.githubusercontent.com/116153815/220617346-a3775aaf-dec5-4d87-990b-8607036c4f11.png)

**Reply by jeromelaban** [reply]
This particular error is a project setup issue, likely cause by duplicate inclusion of XAML files. Make sure to take a look at your csproj.

**Reply by MasterThreadInf** [reply]
You are right indeed.  But still have other issues , please have a look.
https://github.com/MasterThreadInf/UnoDataGridWithExtensionTemplate

**Reply by jeromelaban** [reply]
Can you tell which other issues you're getting?

**Reply by MasterThreadInf** [reply]
![image](https://user-images.githubusercontent.com/116153815/221153234-7596472c-fbe9-4a84-925d-9299e1aeb0a4.png)

**Reply by jeromelaban** [reply]
Thanks. See [this article](https://platform.uno/docs/articles/get-started-wizard.html?tabs=windows#error-netsdk1148-a-referenced-assembly-was-compiled-using-a-newer-version-of-microsoftwindowssdknetdll).

**Reply by MasterThreadInf** [reply]
Windows : That article help to resolve sdk.net issue but new issue appears for which package is needed? or Is there anything else missing?
![image](https://user-images.githubusercontent.com/116153815/221196407-89b00e66-5988-47ed-b7e8-3468a3ef1b25.png)

Wasm : bindablemetadataprovider still shows up.  Where is this namespace even created that too is puzzle and why ?

**Reply by jeromelaban** [reply]
This code is meant to be used with UWP (`Windows.UI.Core.CoreDispatcher`), and you app is using WinUI (`Microsoft.UI.Xaml` in the namespaces at the top). Where did you get this piece of code, it needs to be adjusted to run for WinUI.

**Reply by MasterThreadInf** [reply]
Firstly, I created project using dotnet new unoapp-extensions -o MyProjectName . And then I combined following two solutions, so that I get login and username list in datagrid . Only difference between following two solutions and this template is structure.  Perhaps there need to be added something in that template , not sure. 

https://github.com/unoplatform/Uno.Samples/blob/master/UI/TimeEntry/TimeEntryUno/TimeEntryUno.Shared/Helpers/ErrorDialogHelper.cs
https://github.com/unoplatform/Uno.Samples/tree/master/UI/UnoWCTDataGridSample/WCTTestbed2
